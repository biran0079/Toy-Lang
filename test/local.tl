fun f(n){
  fun g1(){
    local n;
    n=123;
    return n;
  }
  fun g2(){
    n++;
    return n;
  }
  return [g1,g2];
}
l = f(0);
g1 = l[0];
g2 = l[1];
print(g1(),g2(),g2(),g1(),g2());

