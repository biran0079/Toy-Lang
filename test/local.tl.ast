[stmts, [fun, [id, f], [ids, [id, n]], [stmts, [fun, [id, g1], [ids], [stmts, [local, [ids, [id, n]]], [=, [id, n], [int, 123]], [return, [id, n]]]], [fun, [id, g2], [ids], [stmts, [++, [id, n]], [return, [id, n]]]], [return, [list, [id, g1], [id, g2]]]]], [=, [id, l], [call, [id, f], [exps, [int, 0]]]], [=, [id, g1], [list_access, [id, l], [int, 0]]], [=, [id, g2], [list_access, [id, l], [int, 1]]], [call, [id, print], [exps, [call, [id, g1], [exps]]]], [call, [id, print], [exps, [call, [id, g2], [exps]]]], [call, [id, print], [exps, [call, [id, g2], [exps]]]], [call, [id, print], [exps, [call, [id, g1], [exps]]]], [call, [id, print], [exps, [call, [id, g2], [exps]]]]]
