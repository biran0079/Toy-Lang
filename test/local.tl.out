123 1 2 123 3
